app/models/requests/scan.rb

```diff
@@ -21,13 +21,14 @@ def send_approval_status!
     RequestStatusMailer.request_status_for_scan(self).deliver_later if notification_email_address.present?
   end
 
+  # Returns true if a background job was enqueued.
   def submit!
-    SubmitIlliadRequestJob.perform_later(id)
-
-    # This ensures that only scan rules with a destination get sent to the ILS.
-    # We no longer want to send SAL3 requests to the ILS as this is handled by the ILLiad integration.
-    # SAL1/2 requests still go to the ILS at this time.
-    send_to_ils_later! if scan_destination&.dig(:patron_barcode).present?
+    SubmitIlliadRequestJob.perform_later(id).tap do
+      # This ensures that only scan rules with a destination get sent to the ILS.
+      # We no longer want to send SAL3 requests to the ILS as this is handled by the ILLiad integration.
+      # SAL1/2 requests still go to the ILS at this time.
+      send_to_ils_later! if scan_destination&.dig(:patron_barcode).present?
+    end
```

Comment:
I wonder if this captures the intent better (make sure we enqueue all the background jobs we need to):

```suggestion
    result = SubmitIlliadRequestJob.perform_later(id)
    return result if !result || scan_destination&.dig(:patron_barcode).blank?

    # This ensures that only scan rules with a destination get sent to the ILS.
    # We no longer want to send SAL3 requests to the ILS as this is handled by the ILLiad integration.
    # SAL1/2 requests still go to the ILS at this time.
    send_to_ils_later!
```

Reply:
this is a little clearer imo, but the comment should go with the new `return` line, right? also `return result if !result` is kind of...weird and might require some more explanation